expression/integration_test.go

```diff
@@ -4677,3 +4677,15 @@ func (s *testIntegrationSuite) TestFuncCaseWithLeftJoin(c *C) {
 
 	tk.MustQuery("select t1.id from kankan1 t1 left join kankan2 t2 on t1.id = t2.id where (case  when t1.name='b' then 'case2' when t1.name='a' then 'case1' else NULL end) = 'case1' order by t1.id").Check(testkit.Rows("1", "2"))
 }
+
+func (s *testIntegrationSuite) TestIssue11309(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec(`drop table if exists t;`)
+	tk.MustExec(`CREATE TABLE t (a decimal(6,3),b double(6,3),c float(6,3));`)
+	tk.MustExec(`INSERT INTO t VALUES (1.100,1.100,1.100);`)
```

Comment:
Can we add more complicated cases?

Reply:
Any Idea...I think type covertion is enough.

Reply:
decimal(11,7): `123.9999999`, `-123.9999999`

Reply:
double(11, 7)
float(11, 7)

Reply:
done